docs/source/plugin/privileges.rst

```diff
@@ -0,0 +1,172 @@
+===============
+User privileges
+===============
+
+IRC users can have privileges in a **channel**, given by MODE messages such as:
+
+.. code-block:: irc
+
+    MODE #example +ov Nickname Nickname
+
+This will give both OP and Voice privileges to the user named "Nickname" in the
+"#example" channel (and only in this channel). When Sopel receives a MODE
+message it registers and updates its knowledge of a user's privileges in a
+channel, which can be used by plugins in various way.
+
+Access rights
+=============
+
+Privileged users
+----------------
+
+A plugin can limit who can trigger its callables using the
+:func:`~sopel.plugin.require_privilege` decorator::
+
+    from sopel import plugin, privileges
+
+    @plugin.require_privilege(privileges.OP)
+    @plugin.require_chanmsg
+    @plugin.command('chanopcommand')
+    def chanop_command(bot, trigger):
+        # only a channel operator can use this command
+
+This way, only users with OP privileges or above in a channel can use the
+command ``chanopcommand`` in that channel: other users will be ignored by the
+bot. It is possible to tell these users why with the ``message`` parameter::
+
+    @plugin.require_privilege(privileges.OP, 'You need +o privileges.')
+
+.. note::
+
+    A command that requires channel privileges will probably be used for
+    channels only. For that purpose, you should use the
+    :func:`sopel.plugin.require_chanmsg` decorator.
+
+The bot is a user too
+---------------------
+
+Sometimes, you may want the bot to be a privileged user in a channel to allow a
+command. For that, there is the :func:`~sopel.plugin.require_bot_privilege`
+decorator::
+
+    @plugin.require_bot_privilege(privileges.OP)
+    @plugin.require_chanmsg
+    @plugin.command('opbotcommand')
+    def change_topic(bot, trigger):
+        # only if the bot has OP privileges
+
+This way, this command cannot be used if the bot doesn't have the right
+privileges in the channel where it is used, independent from the privileges of
+the user who invokes the command.
+
+As with ``require_privilege``, you can provide an error message::
+
+    @plugin.require_bot_privilege(
+        privileges.OP, 'The bot needs +o privileges.')
+
+And you **can** use both ``require_privilege`` and ``require_bot_privilege`` on
+the same plugin callable::
+
+    @plugin.require_privilege(privileges.VOICE)
+    @plugin.require_bot_privilege(privileges.OP)
+    @plugin.require_chanmsg
+    @plugin.command('special')
+    def special_command(bot, trigger):
+        # only if the user has +v and the bot has +o (or above)
+
+This way, you can allow a less privileged user to access a command for a more
+privileged bot (this works for any combination of privileges).
```

Comment:
Also important here: `require_bot_privilege` will always execute the command if called in PM, because there are no channel privileges. Needs a note about using `require_chanmsg` as appropriate (yes I see it in the examples; still, should have what Furo calls an "admonition").

Reply:
@Exirel are you mucking with the rebase and this will be addressed in the next push? I'm nervous about seeing a resolved conversation with no corresponding change yet.